frontend/app/src/components/accounts/blockchain/Eth2Input.vue

```diff
@@ -1,46 +1,73 @@
 <script setup lang="ts">
 import { onlyIfTruthy } from '@rotki/common';
-import { type PropType } from 'vue';
+import useVuelidate from '@vuelidate/core';
+import {
+  and,
+  helpers,
+  maxValue,
+  minValue,
+  requiredUnless
+} from '@vuelidate/validators';
 import { type Eth2Validator } from '@/types/balances';
+import { type ValidationErrors } from '@/types/api/errors';
 
-const isValid = (percentage: string) => {
-  const perc = Number.parseFloat(percentage);
-  return isFinite(perc) && perc >= 0 && perc <= 100;
-};
+const props = defineProps<{
+  validator: Eth2Validator | null;
+  disabled: boolean;
+  errorMessages: ValidationErrors;
+}>();
 
-const props = defineProps({
-  validator: {
-    required: false,
-    type: Object as PropType<Eth2Validator | null>,
-    default: null
-  },
-  disabled: {
-    required: false,
-    type: Boolean,
-    default: false
-  }
-});
-
-const emit = defineEmits(['update:validator']);
-const { validator } = toRefs(props);
+const emit = defineEmits<{
+  (e: 'update:validator', validator: Eth2Validator | null): void;
+}>();
+const { validator, errorMessages } = toRefs(props);
 const validatorIndex = ref('');
 const publicKey = ref('');
-const percentage = ref<string>();
-
-const updateValidator = (validator: Eth2Validator | null) => {
-  emit('update:validator', validator);
-};
+const ownershipPercentage = ref<string>();
 
 const updateProperties = (validator: Eth2Validator | null) => {
   validatorIndex.value = validator?.validatorIndex ?? '';
   publicKey.value = validator?.publicKey ?? '';
-  percentage.value = validator?.ownershipPercentage ?? '';
+  ownershipPercentage.value = validator?.ownershipPercentage ?? '';
+};
+
+const { tc } = useI18n();
+
+const rules = {
+  validatorIndex: {
+    requiredUnlessKey: requiredUnless(logicAnd(publicKey))
```

Comment:
I guess we don't need `logicAnd` here?

Reply:
So the `requiredUnless` expects a `Ref<boolean>` and `publicKey` is a `Ref<string>` and `logicAnd()` returns a `ComputedRef<boolean>` so `logicAnd` is used a a shortcut for `computed(() => !!get(publicKey))`